### ISSUE TITLE
Che 7.9.2 Release

### ISSUE THREAD

**Issue by nickboldt** [eclipse-che/che#16229]
### List of pending issues / PRs

- [x] [Docs] Workspaces RAM and CPU requirements #14612
- [ ] [Docs] Add #15900 and #15614 to known issues
- [x] When using digests, container names can be longer than 63 characters: https://github.com/eclipse/che/issues/16288, PR: https://github.com/eclipse/che/pull/16290
- [x] [Test] Fix Java based devfile tests https://github.com/eclipse/che/pull/16203
- [x] Java language server failed to start https://issues.redhat.com/browse/CRW-691
- [x] In disconnected environment, OpenShift OAuth cannot be activated https://issues.redhat.com/browse/CRW-709 (nearly done, ETA Wed)
- [x] When installed on OS 4.3 restricted environments, devfile registry public URL is not set https://issues.redhat.com/browse/CRW-710
- [x] [cherry pick] Use the latest version of yeoman-plugin https://github.com/eclipse/che-theia/pull/666
- [ ] theia git extension [publish branch] "permission denied" in askpass.sh #16084
- [x] [Patch version of the Dashboard #256](https://github.com/redhat-developer/codeready-workspaces/pull/256#) - https://github.com/eclipse/che/pull/16406
- [ ] *(add blockers here)*

### Release status

In parallel, the following releases can be done:

* che-theia,
* che-machine-exec,
* che-devfile registry, then che-plugin-registry (once che-theia and machine-exec are done)
* che-parent, then che-docs, then che as Release Candidate for QE

Then in series:

* che-parent, then che-docs, then che as Release
* che-operator, then chectl

- [x] che-theia  _*(1eaecd5cbf93468a74988ddcd347402fcbf14cbe)*_
- [x] che-machine-exec
- [x] che-plugin-registry _(depends on che-theia, che-machine-exec)_
- [x] che-devfile-registry
- [x] che-parent
- [x] che-docs _(depends on parent)_
- [x] che _(depends on docs and parent)_
- [x] che-operator _(depends on all of the above)_
  - [x] Kubernetes community operator PR
  - [x] OpenShift community operator PR
- [x] chectl _(depends on all of the above)_
- [ ] complete current milestone
  - [ ] move incomplete *deferred* issues to backlog
  - [ ] move incomplete *WIP* issues to next milestone
  - [ ] close completed issues
  - [ ] close milestone

| Owner | Process | Script | CI | Artifact(s) |
| --- | --- | --- | --- | --- |
| @azatsarynnyy| [che-theia](https://github.com/eclipse/che-theia/blob/master/RELEASE.md) | [***MANUAL***](https://github.com/eclipse/che-theia/blob/master/RELEASE.md) | [centos](https://ci.centos.org/job/devtools-che-theia-che-release/) | [`quay.io/eclipse/che-theia`](https://quay.io/eclipse/che-theia) |
| @nickboldt / @mkuznyetsov| [che-machine-exec](https://github.com/eclipse/che-machine-exec/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-machine-exec/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-machine-exec-release/) | [`quay.io/eclipse/che-machine-exec`](https://quay.io/eclipse/che-machine-exec)| 
| @nickboldt / @mkuznyetsov | [che-devfile-registry](https://github.com/eclipse/che-devfile-registry/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-devfile-registry/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-devfile-registry-release/) | [`quay.io/eclipse/che-devfile-registry`](https://quay.io/eclipse/che-devfile-registry)| 
| @nickboldt / @mkuznyetsov / @ericwill| [che-plugin-registry](https://github.com/eclipse/che-plugin-registry/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-plugin-registry/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-plugin-registry-release/) | [`quay.io/eclipse/che-plugin-registry`](https://quay.io/eclipse/che-plugin-registry)| 
| @vparfonov / @mkuznyetsov| [che-parent](https://github.com/eclipse/che/blob/master/RELEASE.md) | [***MANUAL***](https://github.com/eclipse/che/blob/master/RELEASE.md) | [centos](https://ci.centos.org/job/devtools-che-parent-che-release/) | [che-parent](https://search.maven.org/search?q=a:che-parent), [che.depmgt](https://search.maven.org/artifact/org.eclipse.che.depmgt/maven-depmgt-pom) |
| @vparfonov / @mkuznyetsov| [che-docs](https://github.com/eclipse/che/blob/master/RELEASE.md) | [***MANUAL***](https://github.com/eclipse/che/blob/master/RELEASE.md) | [centos](https://ci.centos.org/job/devtools-che-docs-che-release/) | [che-docs](https://search.maven.org/search?q=a:che-docs)
| @vparfonov / @mkuznyetsov| [che](https://github.com/eclipse/che/blob/master/RELEASE.md) | [***MANUAL***](https://github.com/eclipse/che/blob/master/RELEASE.md) | [centos](https://ci.centos.org/job/devtools-che-che-release/) | [che.core](https://search.maven.org/search?q=che.core), [`quay.io/eclipse/che-server`](https://quay.io/eclipse/che-server) |
| @tolusha| [che-operator](https://github.com/eclipse/che-operator/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-operator/blob/master/make-release.sh) | ? | [`quay.io/eclipse/che-operator`](https://quay.io/eclipse/che-operator)| 
| @tolusha| [chectl](https://github.com/che-incubator/chectl/blob/master/RELEASE.md) | [make-release.sh](https://github.com/che-incubator/chectl/blob/master/make-release.sh) | [travis](https://travis-ci.org/che-incubator/chectl) | [chectl releases](https://github.com/che-incubator/chectl/releases)

**Comment by azatsarynnyy** [comment]
The patch for about dialog already included in 7.9.1

**Comment by nickboldt** [comment]
Then please... fix my mistake. Link to the commit. Remove the incorrect entry on this issue and move it to 7.9.1 issue. 

Like this:

![image](https://user-images.githubusercontent.com/227597/76017826-3b195f00-5eed-11ea-80c8-1981c74f545b.png)

If you see something, don't just say something. FIX IT. Make it clear. Cite sources. Prove your work. 

You have the power. Please use it. :D

**Comment by nickboldt** [comment]
slip to 7.11:

- [ ] Store credentials into secrets instead of plain text of CheCluster CR fields #16243

**Comment by dmytro-ndp** [comment]
@nickboldt: can we include fix of [import in Che with TLS](https://issues.redhat.com/browse/CRW-728) in this release: https://github.com/eclipse/che-theia/pull/672 ?

**Comment by mkuznyetsov** [comment]
waiting for QE to proceed with RC testing

**Comment by dmytro-ndp** [comment]
@davidfestal, @tolusha: installation of Multiuse User Eclipse Che 7.9.2.RC on minikube 1.1.1 using chectl:next and che-operator:nightly [had failed](https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/basic-MultiUser-Che-check-e2e-tests-against-k8s/1374/console).

Could you, please, say which version of **chectl** and **che-operator** we should use to check installation of Eclipse Che 7.9.2.RC?

**Comment by nickboldt** [comment]
Have you tried 7.9.1?

**Comment by dmytro-ndp** [comment]
I can try with che-operator:7.9.1. 
The problem is the next fixup are in https://github.com/eclipse/che-operator/commits/7.9.x branch, so we can't check it with che-operator:7.9.1:
- In disconnected environment, OpenShift OAuth cannot be activated https://issues.redhat.com/browse/CRW-709 (nearly done, ETA Wed)
- When installed on OS 4.3 restricted environments, devfile registry public URL is not set https://issues.redhat.com/browse/CRW-710

**Comment by nickboldt** [comment]
Sure but you also can't check it with the 7.10 or 7.11 operator either :D

**Comment by davidfestal** [comment]
For now you can use quay.io/dfestal/che-operator:fix-devfile-registry-public-url 
It is based on 7.9.x + my fixes that I added on the 7.9.x.

But I assume that if you need to continuously test the bits of the `7.9.x` branch we should have a CI that produces docker images from here.

**Comment by dmytro-ndp** [comment]
That is why we need to have che-operator:7.9.2 image, and chectl:7.9.2 which will be used to generate crwctl for CRW 2.1.0.

**Comment by nickboldt** [comment]
Are you suggesting that we should revise the release process so that @tolusha will release the operator and chectl BEFORE che server 7.9.2 is live? 

Would that work?

**Comment by dmytro-ndp** [comment]
Yes - it would be perfect!

**Comment by mkuznyetsov** [comment]
while david provided newer che-operator, here is how you could build newest chectl:

git clone https://github.com/che-incubator/chectl
cd chectl && git checkout 7.9.x
yarn

use compiled binaries from "bin" directory

**Comment by davidfestal** [comment]
we don't need the che-server maven artifacts to release che-operator and chectl
We do need the che-server docker images though (at least to test the released bits, which we do during the release)

**Comment by dmytro-ndp** [comment]
It's okay to have hardcoded `che-server:7.9.2`, `che-keycloak:7.9.2` in `che-operator:7.9.2` for pre-release testing so that we replacing existed tags with `rc` when installing Che by using custom-resource.yaml

**Comment by dmytro-ndp** [comment]
@mkuznyetsov: we can't build chectl from https://github.com/che-incubator/chectl/commits/7.9.x on CRW CCI because there is too old node version:
```
Cloning into 'chectl'...
Switched to a new branch '7.9.x'
Branch 7.9.x set up to track remote branch 7.9.x from origin.
yarn install v1.21.1
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
error execa@4.0.0: The engine "node" is incompatible with this module. Expected version ">=10". Got "8.11.1"
error Found incompatible module.
```

Also I don't see significant changes compare to [chectl:7.9.1](https://github.com/che-incubator/chectl/releases/tag/20200309102927) in https://github.com/che-incubator/chectl/commits/7.9.x. So, IMHO, we can use [chectl:7.9.1](https://github.com/che-incubator/chectl/releases/tag/20200309102927) this time.

**Comment by dmytro-ndp** [comment]
@davidfestal, @nickboldt: Eclipse Che 7.9.2.RC installation failed because the was `tlsSupport: true` set in https://github.com/eclipse/che-operator/blob/7.9.x/deploy/crds/org_v1_che_cr.yaml#L40.
It was unexpected, because there was no such changes in release issue description.

**Comment by dmytro-ndp** [comment]
@mkuznyetsov, @nickboldt, @davidfestal: 
We have managed to run E2E tests against Eclipse Che 7.9.2.RC:
- quay.io/eclipse/che-server:rc
- quay.io/eclipse/che-keycloak:rc
- quay.io/eclipse/che-theia:7.9.2
- quay.io/eclipse/che-theia-endpoint-runtime-binary:7.9.2
- quay.io/eclipse/che-devfile-registry:7.9.2
- quay.io/eclipse/che-plugin-registry:7.9.2
- quay.io/eclipse/che-plugin-artifacts-broker:v3.1.1
- [chectl:7.9.1](https://github.com/che-incubator/chectl/releases/tag/20200309102927)
- quay.io/dfestal/che-operator:fix-devfile-registry-public-url ([it should have required changes](https://github.com/eclipse/che/issues/16229#issuecomment-601212850))
- centos/postgresql-96-centos7:9.6

Current result against Multiuser Eclipse Che 7.9.2.RC without TLS support:
1. Java selenium tests passed on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-multiuser-integration-tests/27/
2. OS OAuth integration tests passed on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-ocp-oauth-test/28/
3. Rolling update tests passed on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-rolling-strategy-test/28/

Also, @davidfestal reported that he had successfully tested fixes related to che-operator:7.9.x on a disconnected OS 4.3 cluster, installed with the restricted environment procedure, with proxies + TLS + selfsigned certificates enabled. Result was a successfully started Java Gradle workspace, with working basic services like terminal, JDTLS, outlines, and commands.

Next tests are in-progress:
1. Happy path tests, Vert.x, Java Maven and Python devfile tests on minikube 1.1.1
2. GitHub integration tests.
3. Testing of Eclipse Che with TLS support.

**Comment by dmytro-ndp** [comment]
Happy path tests, Vert.x, Java Maven and Python devfile tests against Multiuser Eclipse Che 7.9.2.RC on minikube 1.1.1 have passed successfully: https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/che-regular-tests/view/pre-release/job/MultiUser-Che-release-check-e2e-tests-against-k8s/50/

**Comment by dmytro-ndp** [comment]
GitHub integration tests have passed as well.

**Comment by dmytro-ndp** [comment]
It was decided to create Eclipse Che 9.7.2.RC2 with the next fix inside:
- Add a possibility to set product version from the product.json (PR: https://github.com/eclipse/che/pull/16406, issue: https://issues.redhat.com/browse/CRW-673)

**Comment by dmytro-ndp** [comment]
Current result against Multiuser Eclipse Che 7.9.2.RC2 without TLS support:
- Java selenium tests passed on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-multiuser-integration-tests/28/
- OS OAuth integration tests passed on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-ocp-oauth-test/29/
- Rolling update tests passed on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-rolling-strategy-test/29/
- Happy path tests, Vert.x, Java Maven and Python devfile tests passed on minikube 1.1.1 https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/basic-MultiUser-Che-check-e2e-tests-against-k8s/1402/

Version number is been displayed correctly in Dashboard:
![Screenshot from 2020-03-20 20-01-32](https://user-images.githubusercontent.com/1197777/77192755-bed66c80-6ae5-11ea-9f9c-f3e22a268551.png)

If there is nothing else to test, Eclipse Che 7.9.2.RC2 can be release.

**Comment by nickboldt** [comment]
Che 7.9.2 server and docs are live in quay and maven central. ~ 11pm EST Mar 20.

**Comment by tolusha** [comment]
`che-operator` and `chectl` released.
